hpx/lcos/promise.hpp

```diff
@@ -59,223 +58,216 @@ namespace lcos {
     ///////////////////////////////////////////////////////////////////////////
     template <typename Result, typename RemoteResult>
     class promise
-      : public detail::promise_base<
-            Result, RemoteResult, detail::promise_data<Result> >
+      : public detail::promise_base<Result, RemoteResult,
+            detail::promise_data<Result>>
     {
-        typedef detail::promise_base<
-                Result, RemoteResult, detail::promise_data<Result>
-           > base_type;
+        typedef detail::promise_base<Result, RemoteResult,
+            detail::promise_data<Result>>
+            base_type;
 
     public:
-        // Effects: constructs a promise object and a shared state.
+        /// \brief constructs a promise object and a shared state.
         promise()
           : base_type()
-        {}
+        {
+        }
 
-        // Effects: constructs a promise object and a shared state. The
-        // constructor uses the allocator a to allocate the memory for the
-        // shared state.
+        /// \brief    constructs a promise object and a shared state. The
+        ///           constructor uses the allocator a to allocate the memory for the
+        ///           shared state.
```

Comment:
IIRC, Doxygen ignores/skips `\brief` sections spanning more than one line. I think just leaving off the `\brief` directive should be fine, though.

Reply:
No I think, you can find http://www.doxygen.nl/manual/docblocks.html.